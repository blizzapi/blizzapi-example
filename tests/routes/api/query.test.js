const request = require('supertest');
const app = require('../../../src/app');

jest.mock('blizzapi', () => ({
  BlizzAPI: jest.fn().mockImplementation(() => ({
    query: (endpoint) => {
      if (endpoint === 'wrongEndpoint') {
        throw new Error('wrong response');
      }
      return Promise.resolve({
        data: `sample data for ${endpoint}`,
      });
    }
  })),
}));

describe('/query/:region/?endpoint=:endpoint', () => {
  const route = `/api/query/us?endpoint=/sample/endpoint`;
  const wrongRoute = `/api/query/us?endpoint=wrongEndpoint`;
  test('should return 200', async () => {
    const response = await request(app).get(route);
    expect(response.statusCode).toBe(200);
  });

  test('should match snapshot', async () => {
    const response = await request(app).get(route);
    expect(response.body).toMatchSnapshot();
  });

  test('should send status code 400 in case of error', async () => {
    const response = await request(app).get(wrongRoute);
    expect(response.statusCode).toBe(400);
  });

  test('error response should match snapshot', async () => {
    const response = await request(app).get(wrongRoute);
    expect(response.body).toMatchSnapshot();
  });
});
