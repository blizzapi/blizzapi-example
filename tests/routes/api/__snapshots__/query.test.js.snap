// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/query/:region/?endpoint=:endpoint error response should match snapshot 1`] = `
Object {
  "error": "Failed to receive Battle.net API data",
}
`;

exports[`/query/:region/?endpoint=:endpoint should match snapshot 1`] = `
Object {
  "data": "sample data for /sample/endpoint",
}
`;
